src/ASM/LR/ASMu2D.C

```diff
@@ -2631,3 +2631,30 @@ void ASMu2D::computeBasis (double u, double v, Go::BasisDerivsSf3& bas,
 
   lrspline->computeBasis(u,v,bas,iel);
 }
+
+
+void ASMu2D::getNeighbours (NeighArray& neighs) const
+{
+  std::vector<std::set<int>> neigh(this->getNoElms(true));
+  const LR::LRSplineSurface* lr = this->getBasis(1);
+  for (const LR::Element* m : lr->getAllElements()) {
+    double epsilon = 1e-6;
+    double umid = (m->umin() + m->umax()) / 2.0;
+    double vmid = (m->vmin() + m->vmax()) / 2.0;
+    std::vector<RealArray> ua = {{m->umin() - epsilon, vmid},
+                                 {m->umax() + epsilon, vmid},
+                                 {umid, m->vmin() - epsilon},
+                                 {umid, m->vmax() + epsilon}};
+    for (const RealArray& u : ua) {
+      int el1 = lr->getElementContaining(u);
+      if (el1 > -1)
+        neigh[m->getId()].insert(el1);
+    }
+  }
+
+  for (size_t i = 0; i < neigh.size(); ++i) {
+    int idx = this->getElmID(i);
```

Comment:
Hup, `getElmID` takes 1-based indices as argument, so `i+1` here (or replace my `MLGE[i]`, the getElmID method is more intended for external use, here and elsewhere in this commit).